### PR TITLE
Cleanup of and performance enhancements to the API installation of repositories.

### PR DESCRIPTION
**Pull request by davebx**
Done:
- [x] Moved everything to a single page, using jquery and AJAX/AJAJ to update the page contents.
- [x] Added a "shopping cart" for repositories. This feature supports multiple toolsheds and multiple repositories within each. Any single repository can be installed at will, or the whole queue can be sent to galaxy for installation.
- [x] Accidentally improved the performance of API installations.
- [x] Added around one line to make it cooperate seamlessly with the excellent work @mvdbeek did with tool dependency resolution and conda.
- [x] Better layout for the various elements.
- [x] Use tabs to make the repository browsing process navigable.
- [x] Tabs behave like tabs. @carlfeberhard gave me some (more) excellent advice.

Needs discussion:
- [ ] I'm debating whether to keep the three new API endpoints or merge them into one, à la `/api/shed_proxy?shed_endpoint=bler_in_blah`

Example screenshot

### CODE REVIEW

File: lib/galaxy/webapps/galaxy/api/tool_shed_repositories.py [hunk 1]
```diff
@@ -130,25 +192,99 @@ def get_latest_installable_revision( self, trans, payload, **kwd ):
...
+        categories = json.loads( util.url_get( url ) )
+        repositories = []
+        url = util.build_url( tool_shed_url, pathspec=[ 'api', 'repositories' ] )
+        for repo in json.loads( util.url_get( url ) ):
+            repositories.append( dict( value=repo[ 'id' ], label='%s/%s' % ( repo[ 'owner' ], repo[ 'name' ] ) ) )
+        return { 'categories': categories, 'repositories': repositories }
+
+    @expose_api
+    @web.require_admin
+    def shed_category( self, trans, **kwd ):
```
Comment (jmchilton) [inline lib/galaxy/webapps/galaxy/api/tool_shed_repositories.py:217]: Can this expose restricted categories or something? `@web.require_admin` is not what I would expect based on the description. (That isn't to say it is wrong at all - I'm just curious - and maybe a comment about why is in order?).

Comment (davebx) [inline lib/galaxy/webapps/galaxy/api/tool_shed_repositories.py:217]: My thinking was that this is an endpoint (currently) used exclusively for repository installation, which is itself an admin-only function. Thus it made sense to make this endpoint also admin-only.

Comment (jmchilton) [reply lib/galaxy/webapps/galaxy/api/tool_shed_repositories.py:217]: I see - I mistakenly thought this was on the tool shed side - sorry.

**Comment by mvdbeek** [comment]
Awesome, that looks very nice!

**Pull request merged**